### ISSUE TITLE
Getting wrong IMEI number 

### ISSUE THREAD

**Issue by rahul-mobilecoderz** [Hina-Hussain/DeviceInformationPlugin#7]
Unable to get the expected IMEI number .

**Comment by braysonjohn148** [comment]
Did you get the solution.

**Comment by rahulkushwaha-dev** [comment]
> Did you get the solution.

No , I can't find any solutions

**Comment by tazzix** [comment]
Android / iOS don't support access to IMEI number anymore, hence the code in the repo:
`      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
        imeiNumber = getDeviceUniqueID();
`

**Comment by Rajkumarlohar65** [comment]
it gives a unique number of device is it always same for the particular device.

**Comment by tazzix** [comment]
Not necessarily, there is some code trying to use the wide vine DRM to generate a UUID, but even then there is a chance that an app re-install or a device reset will cause in a different ID so using the FIS (firebase installation service) sounds like a bit of a better approach but as times are changing, there is no guaranteed way of tracking a device. This is particularly challenging for apps requiring more security, e.g. fintech. But the privacy risks out weigh the benefits:

https://github.com/Hina-Hussain/DeviceInformationPlugin/blob/68a5127bcb118962810df0de7d0cead299e7fa8c/android/src/main/java/co/creativemind/device_information/DeviceInformationPlugin.java#L114

**Comment by jlestrat** [comment]
Get IMEI to Android 9 (API 28) with the telephonyManager work. This condition is wrong :
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P).
I fork to fix this by 
if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P).

**Comment by ahmed-moha** [comment]
> Get IMEI to Android 9 (API 28) with the telephonyManager work. This condition is wrong : if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P). I fork to fix this by if (Build.VERSION.SDK_INT > Build.VERSION_CODES.P).

so what is the deference between the codes ? above
